### PR TITLE
[Validator] use constructor property promotion in Charset and MacAddress constraints

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Validator/Constraints/Charset.php [hunk 1]
```diff
@@ -26,15 +26,13 @@ final class Charset extends Constraint
...
-    public array|string $encodings = [];
-    public string $message = 'The detected character encoding is invalid ({{ detected }}). Allowed encodings are {{ encodings }}.';
-
-    public function __construct(array|string $encodings = null, string $message = null, array $groups = null, mixed $payload = null, array $options = null)
-    {
-        parent::__construct($options, $groups, $payload);
-
-        $this->message = $message ?? $this->message;
-        $this->encodings = (array) ($encodings ?? $this->encodings);
+    public function __construct(
```
Comment (OskarStark) [inline src/Symfony/Component/Validator/Constraints/Charset.php:29]: Removing the options looks like a Bc Break

File: src/Symfony/Component/Validator/Constraints/MacAddress.php [hunk 2]
```diff
@@ -28,25 +27,16 @@ class MacAddress extends Constraint
...
     ];
 
-    public string $message = 'This is not a valid MAC address.';
-
-    /** @var callable|null */
-    public $normalizer;
+    public ?\Closure $normalizer;
 
     public function __construct(
-        array $options = null,
```
Comment (OskarStark) [inline src/Symfony/Component/Validator/Constraints/MacAddress.php:37]: Same like above

File: src/Symfony/Component/Validator/Constraints/Charset.php (hunk 1 above)
Comment (OskarStark) [reply src/Symfony/Component/Validator/Constraints/Charset.php:29]: Ah it's unreleased, so no Bc break, right?

Comment (xabbuh) [reply src/Symfony/Component/Validator/Constraints/Charset.php:29]: Yes, I only touched constraints added for 7.1.

File: src/Symfony/Component/Validator/Constraints/MacAddress.php [hunk 3]
```diff
@@ -28,25 +27,16 @@ class MacAddress extends Constraint
...
-        string $message = null,
+        public string $message = 'This is not a valid MAC address.',
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Constraints/MacAddress.php]: let's also fix the message : "This value is not", related to #53420

Comment (xabbuh) [reply src/Symfony/Component/Validator/Constraints/MacAddress.php]: done

Comment (xabbuh) [reply src/Symfony/Component/Validator/Constraints/MacAddress.php]: I didn't update the translation files as I think you will be doing that in #53420 and the changes will eventually land in the `7.1`  anyway.

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**